src/ray/raylet/main.cc

```diff
@@ -117,21 +115,6 @@ int main(int argc, char *argv[]) {
 
   RAY_CHECK_OK(gcs_client->Connect(main_service));
 
-  // The system_config is only set on the head node--other nodes get it from GCS.
-  if (head_node) {
-    // Parse the configuration list.
-    std::istringstream config_string(config_list);
-    std::string config_name;
-    std::string config_value;
-
-    while (std::getline(config_string, config_name, ',')) {
-      RAY_CHECK(std::getline(config_string, config_value, ';'));
-      // TODO(rkn): The line below could throw an exception. What should we do about this?
-      raylet_config[config_name] = config_value;
-    }
-    RAY_CHECK_OK(gcs_client->Nodes().AsyncSetInternalConfig(raylet_config));
```

Comment:
I assume the motivation of the change is to avoid raylets read internal config before it is written.

In that case, why can't we just block here with promise?

Reply:
Thanks, I think the configuration should be globally consistent. In this case, it is better to write directly by GCS and then read by all raylets. In this way, there is no need to  to write configuration in the raylet side.